### PR TITLE
Share button for copy

### LINKED ISSUE #1105: Review of current share button
Hi, here is a quick review of the "share" buttons:
image

We have:

- Google+ => but Google+ is dead since since april 2019 (https://plus.google.com/)
- Twitter, Facebook, Pocket, Delicious, Mail =>

I think that we can remove Google+ and instead add a button "copy to clipboard". This new button will copy the link to the original article in clipboard. Then users can paste the link every where : in a document, in a mail, in a messenger ans so on. What do you think ?

### PR DESCRIPTION
**Pull request by MiteyIronPaw**
Fixes #1105 

- Still needs Translations.
- It might be worth making the pop-up at the top have absolute positioning so that it can be seen when the user scrolls down and a high z-index so that mobile users can see it when they have a item open.

Otherwise should be good to go. Tested on Firefox, Chrome both on Linux desktop and android.

### CODE REVIEW

File: public/js/selfoss-base.js [hunk 1]
```diff
@@ -621,6 +621,47 @@ var selfoss = {
...
+
+
+    /**
+     * Copy a string to the users clipboard.
+     * This will deselect anything the user has selected and overwirte their clipboard.
+     *
+     * @param stringToCopy
+     * @return true if no execptions were thrown
+     */
+    copyString: function(stringToCopy) {
```
Comment (jtojnar) [inline public/js/selfoss-base.js]: Note, once we merge #1137, we will be able to use something like https://www.npmjs.com/package/copy-to-clipboard.

File: public/js/selfoss-shares.js [hunk 2]
```diff
@@ -33,6 +33,10 @@ selfoss.shares = {
         this.register('mail', 'e', function(url, title) {
             document.location.href = 'mailto:?body=' + encodeURIComponent(url) + '&subject=' + encodeURIComponent(title);
         });
+        this.register('copy', 'c', function(url) {
+            selfoss.copyString(url);
+            selfoss.ui.showMessage(selfoss.ui._('url_copied'));
```
Comment (jtojnar) [inline public/js/selfoss-shares.js]: We currently do not have `selfoss.ui._`. #1137 introduces `selfoss.ui.translate`, though.

Comment (jtojnar) [reply public/js/selfoss-shares.js]: I retract this comment. Forgot we had this.

File: public/js/selfoss-base.js (hunk 1 above)
Comment (MiteyIronPaw) [reply public/js/selfoss-base.js]: Thoughts on merging #1137 into this branch then blocking it until #1137 has been merged?

Comment (jtojnar) [reply public/js/selfoss-base.js]: 👍Hopefully, it will not take much longer to merge it.

Comment (jtojnar) [reply public/js/selfoss-base.js]: Yay, it is in master now.

**Comment by jtojnar** [comment]
Thanks for the help. I switched to a different clipboard library that is closer to the native version. Also changed the icon to a vector one since #1013 got finally merged.

**Comment by MiteyIronPaw** [comment]
Haha thank you :) , I didn't end up doing much in the end.

**Pull request merged**